src/vs/workbench/contrib/debug/browser/debugHover.ts

```diff
@@ -201,6 +201,7 @@ export class DebugHoverWidget implements IContentWidget {
 		const expression = await this.debugHoverComputer.evaluate(session);
```

Comment:
Can this be done by returning a result value rather than passing in a callback? Also, should the editor hover be shown in the cases above too? It also seems weird to show it when the hover is canceled which this change will do.

Reply:
If I got it right , by returning `result` you mean change the return type of this function to `Promise<void | 'some_other_value'>` ? 
?

Reply:
Actually I got the context of your comment , I created a enum with some statuses that indicates the state of the debug expression.

1. When not available , (when the expression is not available / no session available)
2. When the existing expression did not change (when evaluated in the same range)
3. When it gets cancelled ? (Although I'm not sure from where does it get cancelled)

Only when a debug hover expression is not available should it fallback to editor hover.

wdyt ?